### PR TITLE
Add the first create page spec to use Gutenberg

### PR DESCRIPTION
**Pull request by alisterscott**
This adds a Gutenberg editor spec which adds a password protected page

**Testing Instructions**

Make sure `env BROWSERSIZE=desktop ./node_modules/.bin/mocha specs/wp-gutenberg-page-editor-spec.js`

and `env BROWSERSIZE=mobile ./node_modules/.bin/mocha specs/wp-gutenberg-page-editor-spec.js` works

### CODE REVIEW

**Comment by alisterscott** [comment]
This is blocked by https://github.com/Automattic/wp-calypso/issues/28578

**Comment by alisterscott** [comment]
@Automattic/flowpatrol a review would be useful even though this may still be blocked by the issue raised

File: lib/gutenberg/gutenberg-editor-sidebar-component.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import { By } from 'selenium-webdriver';
+import * as driverHelper from '../driver-helper.js';
+import AsyncBaseContainer from '../async-base-container';
+import * as driverManager from '../driver-manager';
+
+export default class GutenbergEditorSidebarComponent extends AsyncBaseContainer {
+	constructor( driver ) {
+		super( driver, By.css( '.block-editor.gutenberg' ) );
+		this.cogSelector = By.css( '[aria-label="Settings"]:not([disabled])' );
```
Comment (alisterscott) [inline lib/gutenberg/gutenberg-editor-sidebar-component.js]: English string but I couldn't find a unique alternative :(

File: lib/gutenberg/gutenberg-editor-sidebar-component.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		this.closeSelector = By.css( '[aria-label="Close settings"]:not([disabled])' );
```
Comment (alisterscott) [inline lib/gutenberg/gutenberg-editor-sidebar-component.js]: English string but I couldn't find a unique alternative :(

File: lib/gutenberg/gutenberg-editor-sidebar-component.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+			if ( c.indexOf( 'is-toggled' ) > -1 ) {
+				return await driverHelper.clickWhenClickable( driver, this.closeSelector );
+			}
+		}
+	}
+
+	async chooseDocumentSetttings() {
+		return await driverHelper.clickWhenClickable(
+			this.driver,
+			By.css( '[data-label="Document"]' )
```
Comment (alisterscott) [inline lib/gutenberg/gutenberg-editor-sidebar-component.js]: English string but I couldn't find a unique alternative :(

**Comment by JavonDavis** [comment]
Realizing we both creating a sidebar component. https://github.com/Automattic/wp-e2e-tests/pull/1627

**Comment by bsessions85** [comment]
Make that 3 of us

**Comment by JavonDavis** [comment]
@bsessions85 I also have a preview component as well, it basically handles switching the tabs instead switching to iFrame like the classic editor

**Comment by alisterscott** [comment]
Raised here https://github.com/WordPress/gutenberg/issues/11951

**Comment by alisterscott** [comment]
Updated to ignore mobile for now to merge it

**Pull request merged**